src/components/connect.js

```diff
@@ -192,7 +196,14 @@ function connect(mapPropsToRequestsToProps, defaults, options) {
         this._unmounted = true
       }
 
+      shouldComponentUpdate(nextProps, nextState) {
+        return !options.pure ||
+          this.state.data != nextState.data ||
+          (dependsOnProps ? (this._propsDidChange || this.props.children != nextProps.children) : !shallowEqual(this.props, nextProps))
```

Comment:
It does add some complexity (and did confuse me for a minute as you see above), but I guess just leave it in since it's already there and tested.